docs/book/src/topics/api-server-endpoint.md

```diff
@@ -0,0 +1,105 @@
+# API Server Endpoint
+
+This document describes how to configure your clusters' api server load balancer and IP.
+
+### Load Balancer Type
+
+CAPZ supports two load balancer types, `Public` and `Internal`. `Public`, which is also the default, means that your API Server Load Balancer will have a publicly accessible IP address.
+`Internal`, also known as a "private cluster", means that the API Server endpoint will only be accessible from within the cluster's virtual network (or peered VNets).
+
+A `Public` cluster will have an Azure public load balancer load balancing internet traffic to the control plane nodes.
+
+A `Private` cluster will have an Azure internal load balancer load balancing traffic inside the VNet to the control plane nodes.
+
+For more information on Azure load balancing, see [Load Balancer documentation](https://docs.microsoft.com/en-us/azure/load-balancer/load-balancer-overview).
+
+<aside class="note warning">
+
+<h1> Warning </h1>
+
+When creating a workload cluster with `apiServerLB` type `Internal`, the management cluster needs to be in the same VNet, or a peered VNet, as the workload cluster. Otherwise, it will not be able to access the target cluster's api server and the cluster creation will fail.  
+
```

Comment:
maybe add a link to how to use with [bastion](https://docs.microsoft.com/en-us/azure/bastion/bastion-overview)?

Reply:
I'll hold until we have https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/165 and have tested Bastion, I don't want to guide users to use it until we've verified it works well with capz.